flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSourceBuilder.java

```diff
@@ -458,6 +458,29 @@ private boolean maybeOverride(String key, String value, boolean override) {
         return overridden;
     }
 
+    private boolean maybeOverridePartitionDiscovery(String key, String value, boolean override) {
+        boolean overridden = false;
+        String userValue = props.getProperty(key);
+        if (override) {
+            LOG.warn(
+                    String.format(
+                            "Property %s is provided but will be overridden from %s to %s",
+                            key, userValue, value));
+            props.setProperty(key, value);
+            overridden = true;
+        } else {
+            if (userValue != null) {
+
```

Comment:
if statement do nothing looks weird.

Reply:
@zhuxiaoshang .thanks for review. Yes, I should use a more elegant way.

Reply:
Imitate another method  maybeoverride . set a default value

Reply:
maybeoverride and your new method.Only the default value is different.